### ISSUE TITLE
make all failed on Centos 7.6 with gcc 10.1.0

### ISSUE THREAD

**Issue by cupcatCoffee** [facebook/rocksdb#10704]
### Actual behavior
When we compiled with gcc 10.1.0 on Centos7.6,we encountered the following error:
  ```
CC       env/env_basic_test.o
  CC       cache/cache_test.o
  CC       cache/cache_reservation_manager_test.o
  CC       cache/lru_cache_test.o
  CC       cache/compressed_secondary_cache_test.o
In file included from db_stress_tool/expected_state.cc:10:
./db_stress_tool/db_stress_common.h:287:15: error: expected constructor, destructor, or type conversion before ‘(’ token
  287 | DECLARE_uint32(memtable_protection_bytes_per_key);
      |               ^
  CC       db/blob/blob_counting_iterator_test.o
  CC       db/blob/blob_file_addition_test.o
  CC       db/blob/blob_file_builder_test.o
  CC       db/blob/blob_file_cache_test.o
  CC       db/blob/blob_file_garbage_test.o
  CC       db/blob/blob_file_reader_test.o
  CC       db/blob/blob_garbage_meter_test.o
In file included from db_stress_tool/db_stress_test_base.cc:15:
./db_stress_tool/db_stress_common.h:287:15: error: expected constructor, destructor, or type conversion before ‘(’ token
  287 | DECLARE_uint32(memtable_protection_bytes_per_key);
      |               ^
  CC       db/blob/blob_source_test.o
  CC       db/blob/db_blob_basic_test.o
  CC       db/blob/db_blob_compaction_test.o
  CC       db/blob/db_blob_corruption_test.o
In file included from db_stress_tool/db_stress_tool.cc:24:
./db_stress_tool/db_stress_common.h:287:15: error: expected constructor, destructor, or type conversion before ‘(’ token
  287 | DECLARE_uint32(memtable_protection_bytes_per_key);
      |               ^
db_stress_tool/db_stress_test_base.cc: In function ‘void rocksdb::InitializeOptionsFromFlags(const std::shared_ptr<rocksdb::Cache>&, const std::shared_ptr<rocksdb::Cache>&, const std::shared_ptr<const rocksdb::FilterPolicy>&, rocksdb::Options&)’:
db_stress_tool/db_stress_test_base.cc:3079:7: error: ‘FLAGS_memtable_protection_bytes_per_key’ was not declared in this scope
 3079 |       FLAGS_memtable_protection_bytes_per_key;
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  CC       db/blob/db_blob_index_test.o
```

### Steps to reproduce the behavior
make -j64 all

**Comment by riversand963** [comment]
I suspect it's a gflags version issue.
I seem to be using 2.2 and it works. However, using a different version of gflags does fail for the same reason. Let me see if it's ok to use int32 for these cases.

**Comment by riversand963** [comment]
Should be fixed by #10729
